### ISSUE TITLE
flow_from_directory() only gives corresponding images if # of samples < batch size

### ISSUE THREAD

**Issue by TimOgden** [keras-team/keras-preprocessing#210]
I am trying to give my neural network dark images as input and lighter images as output. Here is my directory structure:

```
dataset
--> long
----> long
------> long-1.jpg
------> long-2.jpg
------> long-3.jpg ...
--> short
----> short
------> short-1.jpg
------> short-2.jpg
------> short-3.jpg ...
```

Where long-1.jpg is the brighter image of short-1.jpg but looks the same otherwise, etc.

I followed the documentation and the example of transforming images and masks together from the [ImageDataGenerator class](https://keras.io/preprocessing/image/#imagedatagenerator-class) here:

````
seed = 1
short_generator = self.datagen.flow_from_directory('../screenshots/shortext/', class_mode=None,
		target_size=(self.x_res,self.y_res), subset='training', batch_size=self.batch_size, seed=seed)
long_generator = self.datagen.flow_from_directory('../screenshots/longext/', class_mode=None,
		target_size=(self.x_res,self.y_res), subset='training', batch_size=self.batch_size, seed=seed)
generator = zip(short_generator, long_generator)
````

However, if I get the elements of next(generator) and loop through them, I can see that I am getting tuples of a short image and a long image, but they do not correspond. For example, I may get short-1.jpg and long-50.jpg. This is a real issue.

To debug, I made my dataset really small, 11 images total. Testing it, it seemed that the images corresponded perfectly. Then I increased the total to my current batch size, 128. This also yielded corresponding images. However, if I do the full dataset, none of the images seem to correspond. It's possible that the dataset got messed up somewhere and maybe short-256.jpg got repeated twice or something, so now short-257.jpg->long-256.jpg and short-258.jpg->long-258.jpg, something like that. However, I created all these images through the Unity game engine and they were named automatically, so I can't think why it would make a mistake in the naming process. The alternative is flow_from_directory() falls apart when the dataset is too large. Does anyone have any idea on this? Thanks

**Comment by Dref360** [comment]
I cannot reproduce, can you help me?

```python
import os

import numpy as np
from PIL import Image
from keras.preprocessing.image import ImageDataGenerator

pjoin = os.path.join

cls1_pt = 'tmp/cls1/cls1'
cls2_pt = 'tmp/cls2/cls2'
os.makedirs('tmp', exist_ok=True)
os.makedirs(cls1_pt, exist_ok=True)
os.makedirs(cls2_pt, exist_ok=True)

for i in range(100):
    img = Image.fromarray((np.ones([100, 100, 3]) * i % 255).astype(np.uint8))
    img.save(pjoin(cls1_pt, '{}.png'.format(i)))
    img.save(pjoin(cls2_pt, '{}.png'.format(i)))

flow1 = ImageDataGenerator().flow_from_directory('tmp/cls1', seed=1337, shuffle=True)
flow2 = ImageDataGenerator().flow_from_directory('tmp/cls2', seed=1337, shuffle=True)

assert np.allclose(flow1[0][0], flow2[0][0])

assert np.allclose(next(flow1)[0], next(flow2)[0])

```

**Comment by TimOgden** [comment]
Upon further analysis, I can see that I was missing two images in the "short" directory, which is why when I called ImageDataGenerator.flow_from_directory('short') I was getting 2 less than when I called ImageDataGenerator.flow_from_directory('long'). Not a Keras issue, just my mistake. Thank you though!

**Comment by TimOgden** [comment]
> Upon further analysis, I can see that I was missing two images in the "short" directory, which is why when I called ImageDataGenerator.flow_from_directory('short') I was getting 2 less than when I called ImageDataGenerator.flow_from_directory('long'). Not a Keras issue, just my mistake. Thank you though!

I should specify, when I say I was getting 2 less when I called ImageDataGenerator.flow_from_directory('short'), I mean that what prints to the console when flow_from_directory return a generator was 'Found 4396 images belonging to 1 classes.' instead of 'Found 4398 images belonging to 1 classes.'
